### ISSUE TITLE
Uncaught Error: This TextEditor has been destroyed

### ISSUE THREAD

**Issue by darioquintana** [Glavin001/atom-beautify#535]
Steps:
1. Set beautify on save for ALL languages.
2. Tried to save a HTML file and crushed.

**Atom Version**: 1.0.11
**System**: Mac OS X 10.10.5
**Thrown From**: [atom-beautify](https://github.com/Glavin001/atom-beautify) package, v0.28.11

### Stack Trace

Uncaught Error: This TextEditor has been destroyed

```
At /Applications/Atom.app/Contents/Resources/app.asar/src/display-buffer.js:1097

Error: This TextEditor has been destroyed
  at DisplayBuffer.module.exports.DisplayBuffer.screenPositionForBufferPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/display-buffer.js:1097:15)
  at Marker.module.exports.Marker.getHeadScreenPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/marker.js:179:33)
  at Cursor.module.exports.Cursor.getScreenPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:90:26)
  at Cursor.module.exports.Cursor.getScreenRange (/Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:708:20)
  at Cursor.module.exports.Cursor.autoscroll (/Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:713:51)
  at Cursor.module.exports.Cursor.changePosition (/Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:698:21)
  at Cursor.module.exports.Cursor.setBufferPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/cursor.js:97:19)
  at /Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1354:23
  at TextEditor.module.exports.TextEditor.moveCursors (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1593:9)
  at TextEditor.module.exports.TextEditor.setCursorBufferPosition (/Applications/Atom.app/Contents/Resources/app.asar/src/text-editor.js:1353:19)
  at setCursors (/Users/dario/.atom/packages/atom-beautify/src/beautify.coffee:45:14)
  at /Users/dario/.atom/packages/atom-beautify/src/beautify.coffee:439:13

```

### Commands

```
  3x -8:16.6.0 editor:select-to-end-of-word (atom-text-editor.editor.is-focused)
  2x -8:15.8.0 editor:select-to-beginning-of-word (atom-text-editor.editor.is-focused)
     -8:15 core:move-right (atom-text-editor.editor.is-focused)
  4x -8:14.8.0 core:move-left (atom-text-editor.editor.is-focused)
     -8:13.6.0 editor:move-to-beginning-of-word (atom-text-editor.editor.is-focused)
  3x -8:13 editor:move-to-end-of-word (atom-text-editor.editor.is-focused)
     -8:12.2.0 editor:move-to-beginning-of-word (atom-text-editor.editor.is-focused)
  2x -8:11.7.0 editor:select-to-beginning-of-word (atom-text-editor.editor.is-focused)
     -8:05.1.0 core:save (atom-text-editor.editor.is-focused)
     -6:38.8.0 core:paste (atom-text-editor.editor.is-focused)
 16x -6:32.2.0 core:backspace (atom-text-editor.editor.is-focused)
     -6:22.5.0 core:move-up (atom-text-editor.editor.is-focused)
     -6:21.3.0 core:backspace (atom-text-editor.editor.is-focused.autocomplete-active)
  2x -4:58.3.0 key-binding-resolver:toggle (atom-text-editor.editor.is-focused)
     -4:56.1.0 settings-view:open (atom-text-editor.editor.is-focused)
  3x -4:52.2.0 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {},
  "atom-beautify": {
    "_analyticsUserId": "2d45efef-4932-4f91-b7d6-347eac37499a",
    "language_csv_beautify_on_save": true,
    "language_html_beautify_on_save": true,
    "language_json_beautify_on_save": true,
    "language_jsx_beautify_on_save": true,
    "language_markdown_beautify_on_save": true,
    "language_cpp_beautify_on_save": true,
    "language_cs_beautify_on_save": true,
    "language_css_beautify_on_save": true,
    "language_d_beautify_on_save": true,
    "language_ejs_beautify_on_save": true,
    "language_erb_beautify_on_save": true,
    "language_fortran_beautify_on_save": true,
    "language_gherkin_beautify_on_save": true,
    "language_go_beautify_on_save": true,
    "language_handlebars_beautify_on_save": true,
    "language_java_beautify_on_save": true,
    "language_js_beautify_on_save": true,
    "language_less_beautify_on_save": true,
    "language_marko_beautify_on_save": true,
    "language_mustache_beautify_on_save": true,
    "language_objectivec_beautify_on_save": true,
    "language_pawn_beautify_on_save": true,
    "language_perl_beautify_on_save": true,
    "language_php_beautify_on_save": true,
    "language_puppet_beautify_on_save": true,
    "language_python_beautify_on_save": true,
    "language_ruby_beautify_on_save": true,
    "language_rust_beautify_on_save": true,
    "language_sass_beautify_on_save": true,
    "language_scss_beautify_on_save": true,
    "language_spacebars_beautify_on_save": true,
    "language_sql_beautify_on_save": true,
    "language_svg_beautify_on_save": true,
    "language_swig_beautify_on_save": true,
    "language_twig_beautify_on_save": true,
    "language_tss_beautify_on_save": true,
    "language_vala_beautify_on_save": true,
    "language_typescript_beautify_on_save": true,
    "language_visualforce_beautify_on_save": true,
    "language_xml_beautify_on_save": true
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.28.11
file-icons, v1.6.9
minimap, v4.13.2

# Dev
No dev packages
```

**Comment by Glavin001** [comment]
This is strange. It looks like it has something to do with setting the cursor positions back to where they were before beautification occurred. See https://github.com/Glavin001/atom-beautify/blob/master/src/beautify.coffee#L45 

Could you provide more information? Maybe sample code, any steps you may have missed, etc.

**Comment by esetnik** [comment]
@Glavin001 i'm not sure if this helps. I can reproduce this error consistently when using atom with left-right split panes.

Steps to reproduce:
1. Open two JS files in a single pane
2. cmd+k -> to split horizontal right
3. The file that moved right is now open in both panes
4. Close the copy of the file in the left pane
5. Now the file is only open in the right pane
6. Save the file (triggering beautification)

**Comment by esetnik** [comment]
![texteditor has been destroye](https://cloud.githubusercontent.com/assets/664434/9704753/6b147a28-547f-11e5-8285-09bb5e14dbff.png)

**Comment by jbubsk** [comment]
I get this error when I changed file(use CoffeeScript) and close tab without saving

**Comment by Agnostics** [comment]
Any file I split to the right and save(Auto beautification) I get this same error.

**Comment by janwerkhoven** [comment]
+1 having the same issue ([Gist error report](https://gist.github.com/janwerkhoven/12b08744b53765c6e57d))

**Comment by Glavin001** [comment]
Unfortunately, I do not have much time to investigate this bug for the next week or so. If someone is able to submit a Pull Request resolving this bug I will be more than happy to find time to review it, merge, and publish a new release. Thank you!

**Comment by osbornm** [comment]
I'm seeing this pretty much all the time.

**Comment by Glavin001** [comment]
@osbornm are these the steps that you can use to reproduce? If these are the same confirmed steps by multiple users, that helps to narrow it down. 
If no one else has submitted a Pull Request by then, I'll try to take a look after Monday of next week. Thank you everyone for your patience.

**Comment by osbornm** [comment]
I'm away from my desk right now but it always happens on my gulpfile.coffee. In this case I do have a split but the file is open on the left pane. When I get back into the office (Monday) I can get you better repel steps.

**Comment by osbornm** [comment]
Okay the repo for me seems to be split the file, any file type, left or right doesn't matter just so long as it is open in both places. Then close one of them doesn't matter which.  Then when you try to save the file in the remaining window you will get this error from then on. You have to close the atom window, not the pane, to get it to stop.

**Comment by janwerkhoven** [comment]
:+1:  that's at least a temporary workaround

**Comment by Glavin001** [comment]
Could those experiencing this problem please try the Pull Request at https://github.com/Glavin001/atom-beautify/pull/660 and let me know if it works. I'll look into getting the Travis CI tests passing and if that all works then we can merge and resolve this issue. Thanks!

**Comment by Glavin001** [comment]
Published to v0.28.18
